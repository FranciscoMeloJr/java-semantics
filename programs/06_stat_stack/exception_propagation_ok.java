// Testing that exception propagates through method calls

public class exception_propagation_ok {

  public static void main(String[] args) {
    try {
      try {
        throwEx();

        System.out.println("No exception");
      } finally {
        System.out.println("finally after uncaught exception");
      }
    } catch (RuntimeException e) {
      System.out.println("caught exception: "+e);
    }
    System.out.println("Done!");
  }

  static void throwEx() {
    throw new RuntimeException("");
  }
}

// finally after uncaught exception
// caught exception: RuntimeException
// Done!
