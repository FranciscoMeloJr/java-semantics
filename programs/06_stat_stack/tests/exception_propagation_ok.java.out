finally after uncaught exception
caught exception: java.lang.RuntimeException: 
Done!
